platforms/software/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationSpec.groovy

```diff
@@ -642,12 +642,13 @@ class DefaultConfigurationSpec extends Specification {
 
         then:
         // This is not desired behavior. Role should be same as detached configuration.
```

Comment:
I think we can remove this `not desired behavior` comment. 

This was from the before time when we tried to copy the original role of the configuration to the copied configuration. We later made the change to always copy as a `RESOLVABLE_DEPENDENCY_SCOPE`. 

I think the current behavior shown here is desired.

Reply:
I don't think it's desired that a copy ends up with a different role than the original; or that it ends up as a detached configuration.  This is a detail of our weird copy implementation.

I'll expand on this copy to add clarity, but it's worth pointing out, as someone without internal knowledge would read this test and assume something is wrong when the copy varies from the original.

Reply:
Yeah the name of this method is quite bad compared to what its behavior does. According to the behavior it should be called more like `copyAsDetached`. 

Though we have spent a lot of time discussing this already. I don't think we would ever go to make this copy _not_ as a detached configuration, because the use cases for this method are already very questionable. I wouldn't go and try to expand its scope to non-resolvable configurations. 

For that reason, I would say the desired behavior of this method, by definition, is to support whatever existing use cases/behaviors this method already has -- which we determined to be 95% of the case -- to by copying a resolvable configuration.